components/CatModal/CatModalHeader.tsx

```diff
@@ -20,91 +21,144 @@ export default function CatModalHeader({
+              onClick={() => {
+                setDeletePressed(false);
+                setModal({ ...modal, type: 'VIEW_CAT' });
+                modal.type === 'EDIT_CAT' ? updateCat() : sterilizeCat();
+              }}
+              size='xl'
+              radius='md'
+              color='dark'
+              variant='outline'>
+              <SaveIcon />
+            </ActionIcon>
+          )}
+          {modal.type === 'ADD_CAT' && (
+            <ActionIcon
+              onClick={() => {
+                setDeletePressed(false);
+                setModal(modal.back);
+                addCat();
+              }}
+              size='xl'
+              color='dark'
+              radius='md'
+              variant='outline'>
+              <SaveIcon />
+            </ActionIcon>
+          )}
 
-        {modal.type !== 'ADD_CAT' && (
-          <ActionIcon
-            size='xl'
-            onClick={() => {
-              deleteCat();
-              setModal(modal.back);
-            }}
-            radius='md'
-            variant='outline'
-            color='dark'>
-            <Trash size={40} />
-          </ActionIcon>
-        )}
+          {modal.type !== 'ADD_CAT' && (
+            <ActionIcon
+              size='xl'
+              onClick={() => {
+                setDeletePressed(true);
+                setConfirmationVisible(true);
+              }}
+              radius='md'
+              variant='outline'
+              color='dark'>
+              <Trash size={40} />
+            </ActionIcon>
+          )}
+        </Group>
       </Group>
-    </Group>
+      <Collapse in={confirmationVisible}  transitionDuration={500} transitionTimingFunction='ease'>
+        <Stack sx={{backgroundColor: '#C393B0', width: '100%', marginTop: '2px'}} spacing='xs'>
+          {!deletePressed && (<Text align='center' color='white' size='xs' mt='sm'>
+            {t('components.catModal.catModalHeader.confirmation.discardMessage')}
+            </Text>)}
+          {deletePressed && (<Text align='center' color='white' size='xs' mt='sm'>
+            {t('components.catModal.catModalHeader.confirmation.deleteMessage')}
+          </Text>)}
+          <Group align='center' position='center' pb='md'>
+              <Button 
+                onClick={
+                  () => {
+                    setDeletePressed(false);
+                    setConfirmationVisible(false);
+                  }
+                }
+                radius='md' 
+                variant='outline' 
+                sx={{color: '#FFFFFF', borderColor: '#FFFFFF', fontSize: '14px', width: '140px'}}>
+                  {t('components.catModal.catModalHeader.confirmation.cancel')}
+              </Button>
+              <Button 
+                onClick={
+                  () => {
+                    if (deletePressed) {
+                      deleteCat();
+                    }
+                    (modal.back ? setModal(modal.back) : setModal(undefined))
```

Comment:
The `undefined` check seems redudant, please check if code can be reduced to
`setModal(modal.back);`